skorch/_doctor.py

```diff
@@ -0,0 +1,936 @@
+
+        match_fn : callable or None (default=None)
+          If not ``None``, this should be a callable/function that takes the
+          name of a parameter as input and returns a bool as output, where
+          ``False`` indicates that this layer should be excluded. Use this to
+          filter only specific parameters you want to plot.
+
+        axes : np.ndarray of AxesSubplot or None (default=None)
+          By default, a new matplotlib plot is created. If you instead want to
+          plot onto an existing plot, pass it here. There should be one subplot
+          for each top level module (typically 2).
+
+        bins : np.ndarray or None (default=None)
+          Bins to use for the histogram. If left as ``None``, they are inferred
+          from the data.
+
+        **kwargs
+          You can override remaining plotting arguments like ``lw`` (line width)
+          or ``figsize`` (figure size).
+
+        Returns
+        -------
+        axes : np.andarray of AxesSubplot
+          The axes of the plot.
+
+        """
+        self.check_is_fitted()
+
+        # only use modules for which the values are not simply empty
+        module_names = [
+            key for key, val in self.gradient_logs_.items()
+            if any(d for l in val for d in l)
+        ]
+
+        axes = self._get_axes(axes, figsize=figsize, nrows=len(module_names))
+
+        for module_name, ax in zip(module_names, axes):
+            ax = ax[0]  # only 1 col
+            gradients = self.gradient_logs_[module_name][step]
+            if match_fn:
+                gradients = {k: v for k, v in gradients.items() if match_fn(k)}
+                if not gradients:
+                    msg = (
+                        "No parameter found matching the specification of match_fn. "
+                        "Use doctor.get_param_names() to check all parameters."
+                    )
+                    raise ValueError(msg)
+
+            if bins is None:
+                bin_min = min(g.min() for g in gradients.values())
+                bin_max = max(g.max() for g in gradients.values())
+                bins = np.linspace(bin_min, bin_max, 50)
+
+            for key, val in gradients.items():
+                if val.ndim:  # don't show scalars
+                    ax.hist(
+                        val.flatten(),
+                        label=key,
+                        histtype=histtype,
+                        lw=lw,
+                        bins=bins,
+                        density=density,
+                        **kwargs
+                    )
+
+            ax.legend()
+            ax.set_title(f"distribution of gradients for {module_name}")
+
+        return axes
+
+    def plot_param_updates(self, match_fn=None, axes=None, figsize=None, **kwargs):
+        """Plot the distribution of relative parameter updates.
+
+        Plots the log10 of the standard deviation of the parameter update
```

Comment:
I may have missed it, but what was the reason behind `log10` compared to natural log (`ln`)?